### DISCUSSION TITLE
Is there a way to escape {value} in set?

### DISCUSSION THREAD

**Discussion by kujtimiihoxha** [sensedeep/dynamodb-onetable#346] [Q&A] [answered]
Currently when you do something like 

```
await User.update({id: userId}, {set: {title: 'some user adds a title with {abc}'}})
```

This will fail... the main reason is that the logic thinks that you have a variable `{abc}` but I want the logic to treat this as a normal string.

Is there a way to escape `{abc}` and sanitize the user input?

**Comment by kujtimiihoxha** [comment]
Ok for whoever has the same issue later one solution for this is to regex check if the string has any of the template expressions and wrap it all in one expression:

```js
import {nanoid} from 'nanoid'
const myTitle = 'some user adds a title with {abc}'
const substitutions = {}
const setParams = {}
if (typeof myTitle === 'string' && myTitle.match(/\${.*?}|@{.*?}|{.*?}/)) {
	const idKey = nanoid(4)
	substitutions[idKey] = myTitle
	setParams[title] = `@{${idKey}}`
} else {
	setParams[title] =myTitle
}
await User.update({id: userId}, {set: setParams, substitutions})

```

**Comment by mobsense** [comment, marked as answer]
That is not currently possible, but a really good idea to support.

We should allow back quoting like \${var} and \{value}.

To fix, we'd need to change the RegExp in Expression.js:expand and prepareKeyValue to only match if not prefixed with a \ character..

I'll convert this to a bug issue

**Comment by kujtimiihoxha** [comment]
@mobsense yeah that would be best... for now I am using that fix I provided above and it seems to be working as expected
